### PR TITLE
Delete appointment request

### LINKED ISSUE #42: DELETE request of mentee appointments
#### Description
Deletes an appointment from the database (this will be used when a mentor rejects an appointment)

#### Files
`backend/api/views/appointment.py`
(Create if file does not exist yet)

#### Tasks
- [ ] Determine request name, params if necessary, etc.
- [ ] Write the request!
- [ ] Test it on Postman (check if there's anything in the db first with MongoDB Compass for expected results, and make sure it's not breaking the backend!)
- [ ] Create a PR with ^ as screenshots and a description of what you've completed!

### PR DESCRIPTION
**Pull request by nroy20**
## Description :star2:

Created a DELETE request that removes a preexisting appointment request from a mentor.
Fixes #42
 
#### GET request before deletion:
image
#### DELETE request:
image
#### GET request after deletion:
image

## TODOs :star:
- [x] Tests
- [ ] Review

### CODE REVIEW

**Review by AzureQDragon — APPROVED** [review]
Looks good!

File: backend/api/views/appointment.py [hunk 1]
```diff
@@ -6,6 +6,25 @@
 
 appointment = Blueprint("appointment", __name__)
 
+# GET request for appointments by mentor id
+@appointment.route("/appointment/mentor/<string:mentor_id>", methods=["GET"])
+def get_requests_by_mentor(mentor_id):
+    requests = AppointmentRequest.objects(mentor_id=mentor_id)
+    return create_response(data={"requests": requests})
```
Comment (chyku) [inline backend/api/views/appointment.py:13]: we should wait until the get request gets merged in, merge master into this so you can resolve this bc it's not supposed to be here

File: backend/api/views/appointment.py [hunk 2]
```diff
@@ -6,6 +6,25 @@
...
+@appointment.route("/appointment/<string:appointment_id>", methods=["DELETE"])
+def delete_request(appointment_id):
+    try:
+        request = AppointmentRequest.objects.get(id=appointment_id)
+        request.delete()
+    except:
+        msg = "The request you attempted to delete was not found"
+        logger.info(msg)
+        return create_response(status=422, message=msg)
+    return create_response(data={"deleted id": appointment_id})
```
Comment (chyku) [inline backend/api/views/appointment.py:26]: there's no need to return the deleted appt id, the frontend probably wouldn't use it for anything

**Review by chyku — CHANGES_REQUESTED** [review]

File: backend/api/views/appointment.py [hunk 3]
```diff
@@ -72,3 +72,16 @@ def put_appointment(id):
...
+@appointment.route("/appointment/<string:appointment_id>", methods=["DELETE"])
+def delete_request(appointment_id):
+    try:
+        request = AppointmentRequest.objects.get(id=appointment_id)
+    except:
+        msg = "The request you attempted to delete was not found"
+        logger.info(msg)
+        return create_response(status=422, message=msg)
+    request.delete()
+    return create_response(status=200, message=f"Success")
```
Comment (chyku) [inline backend/api/views/appointment.py:87]: you don't need to specify the status if it's a 200, that's the default in [the defintion of create_response](https://github.com/hack4impact-uiuc/mentee/blob/master/backend/api/core.py#L28)

**Review by chyku — APPROVED** [review]

**Pull request merged**